### ISSUE TITLE
Cannot connect Digital Piano to Crouton

### ISSUE THREAD

**Issue by simonsnitz** [dnschneid/crouton#2383]
Hello. My digital piano (Yamaha P-115) cannot be recognized by crouton. It has a USB-Host connection.

When I start up xfce4 and then plug the piano in, I get:
thunar-volman: Unsupported USB device type.
thunar-volman: Unsupported USB device type.
thunar-volman: Unsupported USB device type.

Running dmesg|tail    i get:
[ 4636.719503] sd 5:0:0:0: [sdb] Attached SCSI removable disk
[ 4637.327829] FAT-fs (sdb1): Volume was not properly unmounted. Some data may be corrupt. Please run fsck.
[ 4637.483040] cros-ec-dev cros-ec-dev.1: cros_ec_pd_get_polarity: port:0 enabled:1, role:0, polarity:0, state:2
[ 4637.499273] cros-ec-dev cros-ec-dev.1: cros_ec_pd_get_polarity: port:1 enabled:1, role:0, polarity:0, state:8
[ 5325.509474] usb 1-1: new full-speed USB device number 12 using xhci_hcd
[ 5325.676335] usb 1-1: New USB device found, idVendor=0499, idProduct=160f
[ 5325.676358] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[ 5325.676376] usb 1-1: Product: Digital Piano
[ 5325.676388] usb 1-1: Manufacturer: Yamaha Corporation
[ 5332.824639] haswell-pcm-audio haswell-pcm-audio: FW loaded, mailbox readback FW info: type 01, - version: 00.00, build 77, source commit id: 876ac6906f31a43b6772b23c7c983ce9dcb18a19

       (btw, I also have an external SD card in)

running     df -TH    i get:
Filesystem                                                   Type      Size  Used Avail Use% Mounted on
/dev/root                                                    ext2      1.3G  1.2G  163M  88% /
devtmpfs                                                     devtmpfs  4.2G     0  4.2G   0% /dev
tmp                                                          tmpfs     4.2G  5.6M  4.2G   1% /tmp
run                                                          tmpfs     4.2G  443k  4.2G   1% /run
shmfs                                                        tmpfs     4.2G  266M  3.9G   7% /dev/shm
/dev/sda1                                                    ext4       27G   16G   11G  60% /home
/dev/sda8                                                    ext4       13M   25k   12M   1% /usr/share/oem
/dev/mapper/encstateful                                      ext4      8.0G  124M  7.9G   2% /var
media                                                        tmpfs     4.2G     0  4.2G   0% /media
none                                                         tmpfs     4.2G     0  4.2G   0% /sys/fs/cgroup
/home/.shadow/2eec1a288772e29debdce21364e95c278b71ce76/vault ecryptfs   27G   16G   11G  60% /home/chronos/user
tmpfs                                                        tmpfs     132k   13k  119k  10% /run/crw
/dev/sdb1                                                    vfat       64G  1.6G   63G   3% /media/removable/UNTITLED

    (so I don't believe it is mounted).
what steps must I take to get crouton to connect to and recognize this Digital Piano?

**Comment by tista500** [comment]
@simonsnitz,

> My digital piano (Yamaha P-115) cannot be recognized by crouton. It has a USB-Host connection.

> `[ 4636.719503] sd 5:0:0:0: [sdb] Attached SCSI removable disk`

I don't know what Digital-Piano is, but you meant "external fat32 filesystem won't be mounted via thunar file-manager in crouton"?

If not, did you already tried to install gvfs-fuse to access fat/ntfs filesystem via Gtk+ based graphical file-manager? Or can you mount it manually by `mount` and/or `udiskctl` commands?

Regards.

**Comment by simonsnitz** [comment]
I did have gvfs-fuse installed and tried manually mounting, but couldn't find the proper address in /dev/snd/. The piano shows with dmesg|tail and lsusb  , but I couldn't locate it with /sbin/fdisk -l  

However, despite getting those "Unsupported USB device type" warnings, I got the piano working here by following this tutorial: https://wiki.archlinux.org/index.php/USB_MIDI_keyboards 
very useful. just need to use the (aconnect) and (aseqdump) commands a few times.

Thank you for the help.
